VMEncryption/main/check_util.py

```diff
@@ -90,7 +93,84 @@ def is_invalid_lvm_os(self):
                     detected = True
         return detected
 
-    def is_precheck_failure(self):
+    def check_kv_url(self, test_url, message):
+        """basic sanity check of the key vault url"""
+        expected = "https://{keyvault-name}.vault.azure.net/"
+
+        if test_url is None:
+            raise Exception(message + '\nNo URL supplied')
+
+        try:
+            parse_result = urlparse.urlparse(test_url)
+        except:
+            raise Exception(message + '\nMalformed URL: ' + test_url)
+
+        if not (parse_result.scheme.lower() == "https" and \
+                re.match('[a-zA-Z0-9\-]+\.vault\.azure\.net(:443)?$', parse_result.netloc)):
```

Comment:
>re.match('[a-zA-Z0-9\-]+\.vault\.azure\.net(:443)?$', parse_result.netloc)): [](start = 16, length = 76)

This will fail for national cloud:

public const string AzureKeyVaultDnsSuffix = "vault.azure.net";

public const string ChinaKeyVaultDnsSuffix = "vault.azure.cn";

public const string USGovernmentKeyVaultDnsSuffix = "vault.usgovcloudapi.net";

public const string GermanKeyVaultDnsSuffix = "vault.microsoftazure.de";

Reply:
Adding these to list of possible kv urls

Reply:
Maybe add this link in code comments, since these change over time and this hardcoded list will need to be updated every time a new national cloud is added:  https://docs.microsoft.com/en-us/azure/key-vault/key-vault-access-behind-firewall#key-vault-operations